engines/suppliers/app/controllers/suppliers/api/v1/suppliers_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Suppliers
+  module Api
+    module V1
+      class SuppliersController < ApplicationController
+        before_action :load_supplier, only: [:show, :update, :destroy]
+
+        # GET /api/v1/suppliers
+        #
+        def index
+          suppliers = SuppliersFinder.new(supplier_params[:group_id]).find
```

Comment:
Introducing this, we don't add yet another responsibility to the model and we keep the business logic away from the controller. I believe this abstraction is common way to deal with this.